doc/compressed_transactions.md

```diff
@@ -0,0 +1,153 @@
+| LockTime VarInt       | Variable       | An optional VarInt containing the LockTime if its non zero. |
+| Input/Output Metadata | Variable       | A variable-length bitstream padded to a multiple of 8 bits. Contains metadata on the Inputs and Outputs of the Transaction. See section 3.3 and 3.4. |
+| Input Data            | Variable       | For each Input. See Section 3.5. |
+| Output Data           | Variable       | For each Output. See Section 3.6. |
+
+For the four VarInts listed above we could use a more compact bit encoding for these but they are already a fall back for the bit encoding of the Transaction Metadata.
+
+### 3.3 Transaction Metadata
+
+| Name         | Width       | Description |
+| -------------|-------------|-------------|
+| Version      | VarInt Flag | If 0, indicates a future VarInt. Otherwise encodes the transaction version. |
+| Input Count  | VarInt Flag | If 0, indicates a future VarInt. Otherwise encodes the transaction input count. |
+| Output Count | VarInt Flag | If 0, indicates a future VarInt. Otherwise encodes the transaction output count. |
+| LockTime     | 1 bit       | Wether the LockTime has a value. |
+
+### 3.4 Input Metadata
+
+| Name                        | Width       | Description |
+| ----------------------------|-------------|-------------|
+| Taproot Key or Script Spend | 1 Bits      | For Taproot: 0 for keyspend, 1 for scriptspend; for non-Taproot: 0 for p2sh, 1 for p2sh-wpkh. |
+| Standard Hash               | 1 Bits      | Boolean to determine if this Input's Signature Hash Type is standard (0x00 for Taproot, 0x01 for Legacy/Segwit). |
+| Standard Sequence           | VarInt Flag | If 0, indicates a future VarInt. Otherwise encodes the input sequence as follows: 0b01 = 0x00000000, 0b10 = 0xFFFFFFFE, 0b11 = 0xFFFFFFFF. |
+
+### 3.5.1 Output Metadata
+
+| Name                | Width  | Description |
+| --------------------|--------|-------------|
+| Encoded Script Type | 3 Bits | Encoded Script Type. |
+
+#### 3.5.2 Script Type encoding
+
+| Script Type       | Value |
+| ------------------|-------|
+| Uncompressed P2PK | 0b000 |
+| Compressed P2PK   | 0b001 |
+| P2PKH             | 0b010 |
+| P2SH              | 0b011 |
+| P2WSH             | 0b100 |
+| P2WPKH            | 0b101 |
+| P2TR              | 0b110 |
+| Unknown           | 0b111 |
+
+### 3.6 Input Data
+
+| Name                    | Width            | Description |
+| ------------------------|------------------|-------------|
+| Sequence                | Bitstream VarInt | If the Sequence was non-standard then encoded it here as a Bitstream VarInt. |
+| Hash Type               | 1 Byte           | If the Hash Type for this Input was non standard, contains a single byte. |
+| Txid Blockheight        | Bitstream VarInt | If 0 indicates a coinbase input, Otherwise contains the offset from Minimum Blockheight for the Txid. |
```

Comment:
In ddde5c7defae80058c35bcd5fdc27ee57935495f:

I think you mean to say `0 indicates an uncompressed input` rather than `0 indicates a coinbase input`. I also think this field should come first since it's the most important.

Reply:
Fixed grammar, I have fields that a pertinent to both compressed and uncompressed inputs at the top followed by the deciding flag then the specific fields for compressed or uncompressed.